sdk/core/azure-core/azure/core/pipeline/policies/_universal.py

```diff
@@ -178,18 +181,23 @@ class UserAgentPolicy(SansIOHTTPPolicy):
     _ENV_ADDITIONAL_USER_AGENT = 'AZURE_HTTP_USER_AGENT'
 
     def __init__(self, base_user_agent=None, **kwargs):  # pylint: disable=super-init-not-called
-        # type: (Optional[str], bool) -> None
+        # type: (Optional[str], Dict) -> None
         self.overwrite = kwargs.pop('user_agent_overwrite', False)
         self.use_env = kwargs.pop('user_agent_use_env', True)
+        application_id = kwargs.pop('user_agent', None)
+        sdk_moniker = kwargs.pop('sdk_moniker', 'core/{}'.format(azcore_version))
```

Comment:
I don't think we need the default value - and if we do, it should be `azure-core`, not `core`.

Reply:
Currently, we have azsdk-python-core for azure core, azsdk-python-appconfiguration for app config, azsdk-python-keyvault for keyvault and azsdk-python-storage for storage.

Reply:
Wait. The `sdk_moniker` `**kwarg` gets combined with the `azcore_version`?

Reply:
sdk_moniker is "[sdk name]/[version]".
if not specified, we use "core"(the name)/[azcore_version] (version) as default value.

Reply:
@johanste I realized that without that, downstream package like Storage can't really re-use this policy, otherwise they'll have to specify their own Python platform and Python version (which is what Storage does, I didn't check the other). since "core" was hardcoded in the policy.
The "sdk_moniker" is a way to re-use this policy, while providing your own name and version, so storage and others can use it and let the policy deal with platform info. The default string is the backward compat current one.

Reply:
Note to self - don't review things when you are in a rush to get home. I misread the code.